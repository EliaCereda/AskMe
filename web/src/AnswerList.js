import React, {Component} from "react";
import {Link} from "react-router-dom";
import Question from "./Question";
import Answer from "./Answer";

class AnswerList extends Component {
    render() {
        const answerList = this.props.answers.map((answer) =>
            <div className="AnswerItem" key={answer.id}>
                <Answer {...answer} currentUser={this.props.currentUser} onVote={this.props.onVote} />
            </div>
        );

        // Display the most recent questions first.
        answerList.reverse();

        return answerList;
    }
}

export default AnswerList;
