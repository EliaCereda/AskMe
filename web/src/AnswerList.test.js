import React from 'react';
import AnswerList from "./AnswerList";

describe('AnswerList', () => {
    it('renders correctly', () => {
        const answers = [
            {
                id: 1,
                question_id: 1,
                body: 'corpo della risposta',
                user: 'utente',
                _links: {self: 'answers/1'},
                votes: [
                        {
                          "_links": {},
                          "answer_id": 1,
                          "user": "gio",
                          "value": 1
                        },
                        {
                          "_links": {},
                          "answer_id": 1,
                          "user": "fghj",
                          "value": 1
                        }
                ],
                score: 2,
            },
            {
                id: 2,
                question_id: 1,
                body: 'corpo della risposta',
                user: 'utente2',
                _links: {self: 'answers/2'},
                votes: [
                        {
                          "_links": {},
                          "answer_id": 2,
                          "user": "gio",
                          "value": -1
                        },
                        {
                          "_links": {},
                          "answer_id": 2,
                          "user": "fghj",
                          "value": 1
                        }
                ],
                score: 0,
            },
        ];

        const wrapper = shallow(<AnswerList answers={answers} />);
        expect(wrapper).toMatchSnapshot();
    });
});

