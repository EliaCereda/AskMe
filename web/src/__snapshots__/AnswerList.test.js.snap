// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerList renders correctly 1`] = `
Array [
  <div
    className="AnswerItem"
    key="2"
  >
    <Answer
      _links={
        Object {
          "self": "answers/2",
        }
      }
      body="corpo della risposta"
      id={2}
      question_id={1}
      score={0}
      user="utente2"
      votes={
        Array [
          Object {
            "_links": Object {},
            "answer_id": 2,
            "user": "gio",
            "value": -1,
          },
          Object {
            "_links": Object {},
            "answer_id": 2,
            "user": "fghj",
            "value": 1,
          },
        ]
      }
    />
  </div>,
  <div
    className="AnswerItem"
    key="1"
  >
    <Answer
      _links={
        Object {
          "self": "answers/1",
        }
      }
      body="corpo della risposta"
      id={1}
      question_id={1}
      score={2}
      user="utente"
      votes={
        Array [
          Object {
            "_links": Object {},
            "answer_id": 1,
            "user": "gio",
            "value": 1,
          },
          Object {
            "_links": Object {},
            "answer_id": 1,
            "user": "fghj",
            "value": 1,
          },
        ]
      }
    />
  </div>,
]
`;
